relay-filter/src/lib.rs

```diff
@@ -45,9 +46,16 @@ pub fn should_filter(
     client_ip: Option<IpAddr>,
     config: &FiltersConfig,
 ) -> Result<(), FilterStatKey> {
-    // NB: The order of applying filters should not matter as they are additive. Still, be careful
-    // when making changes to this order.
+    // We support the filtering with the new filters, only for the same or older versions.
+    if config.generic.version <= generic_filters::VERSION {
```

Comment:
nit: I would put this comparison in a helper method `is_enabled`, and check it inside `should_filter`. That way, `VERSION` can even become a private constant if I'm not mistaken.

See for example

https://github.com/getsentry/relay/blob/7392b5d502bb760a0baa6800f2876737a06aedf5/relay-dynamic-config/src/metrics.rs#L51-L53